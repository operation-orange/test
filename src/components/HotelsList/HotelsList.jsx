import React from 'react';
import PropTypes from 'prop-types';

const HotelsList = ({ data }) => (
  data.length ? (
    <table className="hotels-list">
      <thead>
        <tr>
          <td>Name</td>
          <td>Rating</td>
          <td>Facilities</td>
        </tr>
      </thead>
      <tbody>
        {data.map(hotel => (
          <tr key={hotel.id}>
            <td>{hotel.name}</td>
            <td>{hotel.rating}</td>
            <td>{hotel.facilities.join(', ')}</td>
          </tr>
        ))}
      </tbody>
    </table>
  ) : (
    <span>No results</span>
  )
);

HotelsList.propTypes = {
  data: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.number.sRequired,
    name: PropTypes.string.sRequired,
    rating: PropTypes.number.isRequired,
    facilities: PropTypes.arrayOf(PropTypes.string),
  })).isRequired,
};

export default HotelsList;
