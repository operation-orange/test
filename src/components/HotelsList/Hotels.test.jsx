import { shallow } from 'enzyme';
import React from 'react';
import HotelsList from './HotelsList';

it('renders empty list', () => {
  const wrapper = shallow(<HotelsList data={[]} />);
  expect(wrapper).toMatchSnapshot();
});

it('renders list', () => {
  const mockData = [{
    id: 1,
    name: 'Testy',
    rating: 5,
    facilities: ['Gym'],
  }, {
    id: 2,
    name: 'Westy',
    rating: 3,
    facilities: [],
  }];
  const wrapper = shallow(<HotelsList data={mockData} />);
  expect(wrapper).toMatchSnapshot();
});
