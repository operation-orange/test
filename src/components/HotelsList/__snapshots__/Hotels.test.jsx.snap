// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders empty list 1`] = `
<span>
  No results
</span>
`;

exports[`renders list 1`] = `
<table
  className="hotels-list"
>
  <thead>
    <tr>
      <td>
        Name
      </td>
      <td>
        Rating
      </td>
      <td>
        Facilities
      </td>
    </tr>
  </thead>
  <tbody>
    <tr
      key="1"
    >
      <td>
        Testy
      </td>
      <td>
        5
      </td>
      <td>
        Gym
      </td>
    </tr>
    <tr
      key="2"
    >
      <td>
        Westy
      </td>
      <td>
        3
      </td>
      <td />
    </tr>
  </tbody>
</table>
`;
